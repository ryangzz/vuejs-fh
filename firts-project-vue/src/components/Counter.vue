<template>
  <h2>{{title}}</h2>
  <p>{{upNumber}}<sup>2</sup></p>
    <button @click="increment">+</button>
    <button @click="decrement">-</button>
  <p data-id="counter">{{counter}}</p>
</template>

<script>
export default {
  name:"counter",
  props:{
    title:{type:String, default:'Hola soy el contador'},
    start:{
      type:Number,
      default:1
    }
  },
  data() {
    return {
      counter:1
    }
  },
  methods: {
    increment(){
      this.counter += 1;
    },
    decrement(){
      this.counter -= 1;
    }
  },
  computed:{
    upNumber(){
      return this.counter * this.counter;
    }
  }
}
</script>

<style>
button {
  background-color: #41b883;
  color: white;
  border: none;
  padding: 8px 16px;
  border-radius: 4px;
  font-size: 16px;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
}

button:hover {
  background-color: #3a8e81;
}
</style>