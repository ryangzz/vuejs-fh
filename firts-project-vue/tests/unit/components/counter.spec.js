import { shallowMount } from "@vue/test-utils";
import Counter from '@/components/Counter.vue';

describe("Counter Component", ()=>{
  let wrapper;

  beforeEach(()=>{
    wrapper = shallowMount(Counter);
  })
  test("Comprobate Snapshot of the component", ()=>{
    expect(wrapper.html()).toMatchSnapshot();
  })
  test("Validate value Counter default", ()=>{
    expect(wrapper.find('h2')).toBeTruthy();
    const h2Value = wrapper.find("h2").text();
    expect(h2Value).toBe("Hola soy el contador");
  })
  test("el valor de counter por defecto de  100", ()=>{
    expect(wrapper.find('[data-id="counter"]')).toBeTruthy();
    const pValue  = wrapper.find('[data-id="counter"]').text();
    expect(pValue).toBe("1");
  })

  test("debe incrementar 1 y decrementar dos el counter", async ()=>{
    const [incrementar, descementar] = wrapper.findAll("button");
    await incrementar.trigger("click");
    let pValue = wrapper.find('[data-id="counter"]').text();
    expect(pValue).toBe("2");
    await descementar.trigger("click");
    await descementar.trigger("click");
    pValue = wrapper.find('[data-id="counter"]').text();
    expect(pValue).toBe("0");

  })

  test("evaluando los props counter", ()=>{
    const { start } = wrapper.props();
    let value = wrapper.find('[data-id="counter"]').text()*1;
    expect(start).toBe(value);
  })

  test("evaluando titulo contexto", ()=>{
    const title = "Hola soy el contador";
    const wrapper = shallowMount(Counter, {
      props:{
        title
      }
    });
    console.log(wrapper.find("h2").text(), title);
    expect(wrapper.find("h2").text()).toBe(title)
  });
});