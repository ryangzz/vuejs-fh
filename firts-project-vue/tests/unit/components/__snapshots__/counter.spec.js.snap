// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Comprobate Snapshot of the component 1`] = `
<h2>Hola soy el contador</h2>
<p>1<sup>2</sup></p>
<button>+</button>
<button>-</button>
<p data-id="counter">1</p>
`;
